chainer/training/triggers/minmax_value_trigger.py

```diff
@@ -0,0 +1,89 @@
+from chainer import reporter
+from chainer.training.trigger import IntervalTrigger
+
+
+class BestValueTrigger(object):
+
+    """Trigger invoked when specific value becomes best.
+
+    Args:
+        key (str): Key of value.
+        compare (function): Compare function which takes current best value and
+        new value and returns whether new value is better than current best.
+
+    """
+    def __init__(self, key, compare):
+        self._key = key
+        self._min_value = None
+        self._interval_trigger = IntervalTrigger(1, 'epoch')
```

Comment:
Currently, `BestValueTrigger` accumulates the target value throughout the epoch and compares the value with the current best value only at the end of each epoch. We can make this trigger more flexible by adding an tuple that determines the comparison interval (e.g. `1, 'epoch'`) and create `IntervalTrigger` with the tuple.